### DISCUSSION TITLE
A Design Guide for GDevelop

### DISCUSSION THREAD

**Discussion by Midhil457** [4ian/GDevelop#3278] [Enhancement ideas (and their technical discussions)]
Hi,
recently I have been trying to start slowing contributing to the GDevelop codebase and somethings have come to my notice. And I will discuss the solution at the end.

### Irregular naming
I felt that the naming/sentences didn't match a style (Descriptions,titles,etc...). Should the descriptions have `.` at the end (The answer is yes afaik). Should it follow second person or first person (Should it be `My profile` or `Your profile`). 
Should there be `?` and should the parameters to considered questions, like this?? Side note, I think making it interrogative makes it look like GDevelop is asking the user. While not having it, makes it feel like the user is commanding GDevelop (Which I think should be preferred)
![irn-1](https://user-images.githubusercontent.com/73597906/142856871-84746ee9-5ab6-4150-95b9-247564cce291.png)

I think you guys get the ideas. The sentences don't really follow any guideline

### Mixed up design
While some parts of the UI has sharp edges like the button, tabs, list items,etc... Some have rounded corners like the dialogs, text boxes, checkboxes, emptyPlaceHolder,etc... 
And sometimes it creates these weird places (The button is sharp while the dialog is rounded, making it look weird)
![Irn-6](https://user-images.githubusercontent.com/73597906/142857897-1d2be3cb-e7b2-4344-bd3d-4893cca555c2.png)

**I am not saying that rounded corners and sharp corners shouldn't exist together, but the way it's done in GDevelop makes it feel that the components don't co-exist**

Some components that do a similar transition/things (Like expanding). In the list, it's snaping without any transitions, but the accordian does a sweet transition

List:

https://user-images.githubusercontent.com/73597906/142859576-fc5e05db-25ff-40bd-877a-7eddcee7ec15.mp4

Accordian:

https://user-images.githubusercontent.com/73597906/142859618-9e890c55-0f5d-4c81-ab57-5cac7d578834.mp4

There are might be many more genaral things that are irregular. This was just a list I thought were the most noticable

## Solution
Having a design guideline might help solve this problem. Yes, there is material design that GDevelop follows. But, GDevelop has a unique style/personality and there needs to be a unique guide. The guide can document the unique style GDevelop has. It can help contributors develop better components and contribute better. A guide explaining and giving the solution of the above problems. All components are designed with the creator's liking/decisions, and this will cause issues like this. We could ask to make changes to match the style used by other components, but different components follow different styles (See the first two heading in this post for examples). If there is a guide, they can be directed to that.

:)

**Comment by 4ian** [comment]
These are all good remarks! I agree we would need to work on the consistency of the design.

To give a few answers:
- We should aim for period at the end of all description/sentences in the app. The only exception are short labels and actions/conditions.
- No `?` probably - the checkbox is giving the hint that it's a choice. I'll fix this now!
- Agree on the rounding, we might want to standardize this or make it customizable in the themes.
